### ISSUE TITLE
Invalid selector if using shell conditional expressions and selectors

### ISSUE THREAD

**Issue by pelson** [conda/conda-build#786]
I tried implementing the following test in my meta.yaml:

```
test:
    commands:
        - "[ $($PREFIX/bin/geos-config --version) == '3.4.2' ]"  # [not win]
```

Turns out, I get an invalid selector error (https://github.com/conda/conda-build/blob/master/conda_build/metadata.py#L84).

Related issues: #698, #629.

**Comment by stuarteberg** [comment]
I know this issue is super-old at this point, but FWIW, this was fixed via #957

**Comment by msarahan** [comment]
Thanks for helping to clean up @stuarteberg!
